aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotProxiedSSLSocket.java

```diff
@@ -0,0 +1,355 @@
+/**
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *    http://aws.amazon.com/apache2.0
+ *
+ * This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.amazonaws.mobileconnectors.iot;
+
+import android.annotation.TargetApi;
+import android.os.Build;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.SocketException;
+import java.nio.channels.SocketChannel;
+
+import javax.net.ssl.HandshakeCompletedListener;
+import javax.net.ssl.SSLParameters;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+
+class AWSIotProxiedSSLSocket extends SSLSocket {
+
+    private final SSLSocketFactory socketFactory;
+    private final Socket proxySocket;
+    private SSLSocket socket;
+
+
+    public AWSIotProxiedSSLSocket(SSLSocketFactory socketFactory, Socket socket) {
+        this.socketFactory = socketFactory;
+        this.proxySocket = socket;
+    }
+    @Override
+    public void connect(SocketAddress socketAddress) throws IOException {
+        connect(socketAddress, 0);
+    }
+
+    @Override
+    public void connect(SocketAddress socketAddress, int timeout) throws IOException {
+        proxySocket.connect(socketAddress, timeout);
+        this.socket = (SSLSocket) socketFactory.createSocket(proxySocket,
+                ((InetSocketAddress) socketAddress).getHostName(), ((InetSocketAddress) socketAddress).getPort(), true);
+    }
+
+    @Override
+    public String[] getSupportedCipherSuites() {
+        return this.socket.getSupportedCipherSuites();
+    }
+
+    @Override
+    public String[] getEnabledCipherSuites() {
+        return this.socket.getEnabledCipherSuites();
+    }
+
+    @Override
+    public void setEnabledCipherSuites(String[] strings) {
+        this.socket.setEnabledCipherSuites(strings);
+    }
+
+    @Override
+    public String[] getSupportedProtocols() {
+        return this.socket.getSupportedProtocols();
+    }
+
+    @Override
+    public String[] getEnabledProtocols() {
+        return this.socket.getEnabledProtocols();
+    }
+
+    @Override
+    public void setEnabledProtocols(String[] strings) {
+        this.socket.setEnabledProtocols(strings);
+    }
+
+    @Override
+    public SSLSession getSession() {
+        return this.socket.getSession();
+    }
+
+    @TargetApi(Build.VERSION_CODES.N)
```

Comment:
What minimum Android API level is required for this annotation to compile? IoT depends on API level 10, so I would recommend checking on a private maven build to ensure this annotation will compile.